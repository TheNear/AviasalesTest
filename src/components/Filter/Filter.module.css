.filter {
  grid-area: filter;
  width: 100%;
  overflow: hidden;
  justify-self: start;
}

.wrapper {
  width: 100%;
  padding: 10px 0;
  border-radius: 5px;
  background-color: #ffffff;
  box-shadow: 0px 2px 8px rgba(0, 0, 0, 0.1); 
  word-wrap: break-word;
}

.title {
  padding: 10px 20px;
  font-weight: 600;
  font-size: inherit;
  text-transform: uppercase;
  letter-spacing: .5px;
}
