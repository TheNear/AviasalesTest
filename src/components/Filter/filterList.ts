export interface IFilterItem {
  text: string;
  name: string;
}

export const filterList = [
  {
    text: "Все",
    name: "filter__all__transfer",
  },
  {
    text: "Без пересадок",
    name: "filter__without__transfer",
  },
  {
    text: "1 пересадка",
    name: "filter__one__transfer",
  },
  {
    text: "2 пересадка",
    name: "filter__two__transfer",
  },
  {
    text: "3 пересадка",
    name: "filter__three__transfer",
  },
];
