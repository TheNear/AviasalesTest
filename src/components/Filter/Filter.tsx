import React from "react";
import style from "./Filter.module.css";
import commonStyle from "../../assets/css/common.module.css";
import CheckBox from "../../componentsHelper/CheckBox/CheckBox";
import { filterList, IFilterItem } from "./filterList";

const Filter: React.FC = () => {
  return (
    <div className={style.filter}>
      <div className={`${style.wrapper} ${commonStyle.box}`}>
        <h3 className={style.title}>Количество пересадок</h3>
        <ul>
          {filterList.map((filter: IFilterItem) => (
            <CheckBox key={filter.name} text={filter.text} name={filter.name} />
          ))}
        </ul>
      </div>
    </div>
  );
};

export default Filter;
