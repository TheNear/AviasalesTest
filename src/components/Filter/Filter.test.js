import React from "react";
import { shallow } from "enzyme";
import Filter from "./Filter";

describe("Test Filter component", () => {
  it("Render empty Filter", () => {
    const wrapper = shallow(<Filter />);
    expect(wrapper).toMatchSnapshot();
  });
});
