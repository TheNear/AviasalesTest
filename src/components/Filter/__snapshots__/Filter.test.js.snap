// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Filter component Render empty Filter 1`] = `
<div
  className="filter"
>
  <div
    className="wrapper box"
  >
    <h3
      className="title"
    >
      Количество пересадок
    </h3>
    <ul>
      <CheckBox
        key="filter__all__transfer"
        name="filter__all__transfer"
        text="Все"
      />
      <CheckBox
        key="filter__without__transfer"
        name="filter__without__transfer"
        text="Без пересадок"
      />
      <CheckBox
        key="filter__one__transfer"
        name="filter__one__transfer"
        text="1 пересадка"
      />
      <CheckBox
        key="filter__two__transfer"
        name="filter__two__transfer"
        text="2 пересадка"
      />
      <CheckBox
        key="filter__three__transfer"
        name="filter__three__transfer"
        text="3 пересадка"
      />
    </ul>
  </div>
</div>
`;
